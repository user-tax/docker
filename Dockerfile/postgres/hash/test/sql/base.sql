CREATE EXTENSION hash;

SELECT hash('A')::hash;
SELECT hash('B')::hash;
SELECT hash('A')::hash = hash('A')::hash;
SELECT hash('A')::hash = hash('B')::hash;

SELECT hash('A')::hash <= hash('B')::hash;
SELECT hash('A')::hash <  hash('B')::hash;
SELECT hash('A')::hash >= hash('B')::hash;
SELECT hash('A')::hash >  hash('B')::hash;

CREATE TABLE test ( id hash );
INSERT INTO test SELECT hash(i::text) FROM generate_series(1,10000) s(i);
ANALYZE test;
SELECT relpages FROM pg_class WHERE relname = 'test';