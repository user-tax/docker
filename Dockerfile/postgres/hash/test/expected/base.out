CREATE EXTENSION hash;
SELECT hash('A')::hash;
               hash                
----------------------------------
 7fc56270e7a70fa81a5935b72eacbe29
(1 row)

SELECT hash('B')::hash;
               hash                
----------------------------------
 9d5ed678fe57bcca610140957afab571
(1 row)

SELECT hash('A')::hash = hash('A')::hash;
 ?column? 
----------
 t
(1 row)

SELECT hash('A')::hash = hash('B')::hash;
 ?column? 
----------
 f
(1 row)

SELECT hash('A')::hash <= hash('B')::hash;
 ?column? 
----------
 t
(1 row)

SELECT hash('A')::hash <  hash('B')::hash;
 ?column? 
----------
 t
(1 row)

SELECT hash('A')::hash >= hash('B')::hash;
 ?column? 
----------
 f
(1 row)

SELECT hash('A')::hash >  hash('B')::hash;
 ?column? 
----------
 f
(1 row)

CREATE TABLE test ( id hash );
INSERT INTO test SELECT hash(i::text) FROM generate_series(1,10000) s(i);
ANALYZE test;
SELECT relpages FROM pg_class WHERE relname = 'test';
 relpages 
----------
       55
(1 row)

